src/components/SignInForm/Form.tsx

```diff
@@ -1,12 +1,35 @@
-import React from 'react'
-import { useRouter } from 'next/router'
+import React, { useEffect, useState } from 'react'
+import { signIn, signOut } from 'next-auth/client'
+import router from 'next/router'
 
 const Form = () => {
-  const router = useRouter()
+  // TODO: Further implement this
+  // const [loginError, setLoginError] = useState('')
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+
+  useEffect(() => {
+    // Getting the error details from URL
+    if (router.query.error) {
+      // setLoginError(router.query.error) // Shown below the input field in my example
+      // setEmail(router.query.email) // To prefill the email after redirect
+    }
+  }, [router])
 
   const handleSubmitClick = (event: React.MouseEvent<HTMLButtonElement>) => {
     event.preventDefault()
-    router.push('/')
+    signIn('credentials',
+    {
+      email: email,
+      password: password,
+      callbackUrl: '/'
+    })
+  }
```

Comment:
Should we use Prettier formatter on this?

```suggestion
    signIn('credentials', {
      email: email,
      password: password,
      callbackUrl: '/'
    })
  }
```